Marlin/src/module/thermistor/thermistors.h

```diff
@@ -25,12 +25,13 @@
 
 #define THERMISTOR_TABLE_ADC_RESOLUTION 1024
 #define THERMISTOR_TABLE_SCALE (HAL_ADC_RESOLUTION / (THERMISTOR_TABLE_ADC_RESOLUTION))
-#define MAX_RAW_TERMISTOR_VALUE (0x4000 - 1)
 #if ENABLED(HAL_ADC_FILTERED)
   #define OVERSAMPLENR 1
 #else
-  #define OVERSAMPLENR ((MAX_RAW_TERMISTOR_VALUE + 1) / (HAL_ADC_RESOLUTION))
+  #define OVERSAMPLENR 16
 #endif
+#define MAX_RAW_TERMISTOR_VALUE ((HAL_ADC_RESOLUTION * OVERSAMPLENR) - 1)
```

Comment:
Currently Marlin will go into kill state if raw value is more than 14-bit. That is why I set it constant and calculated the oversampling depth from it.

Reply:
That was caused because the ADC value was more than MAX_RAW_TERMISTOR_VALUE with how its calculated now that's not an issue

The hard 15bit limit (after oversampling) because of storing all ADC values as `int16_t` is still a problem though.

It was late last night (morning) when I was fixing this up so I'l confirm my interpretation of what was going on.

Reply:
double checked and 15bit ADC readings seems to work perfectly, but I can't check actually heating a hot end right now.